source/dynamical_systems/test/tests/test_point_attractor.cpp

```diff
@@ -31,178 +37,100 @@ class LinearDSTest : public testing::Test {
   double angular_tol = 1e-3;
 };
 
-TEST_F(LinearDSTest, EmptyConstructorCartesianState) {
+TEST_F(PointAttractorTest, EmptyConstructorCartesianState) {
   // construct empty cartesian state DS
   CartesianPose attractor = CartesianPose::Identity("CAttractor", "A");
-  Linear<CartesianState> ds;
+
   // base frame and attractor should be empty
-  EXPECT_TRUE(ds.get_attractor().is_empty());
-  EXPECT_TRUE(ds.get_base_frame().is_empty());
-  ds.set_attractor(attractor);
-  EXPECT_FALSE(ds.get_attractor().is_empty());
-  EXPECT_FALSE(ds.get_base_frame().is_empty());
+  EXPECT_TRUE(ds->get_parameter_value<CartesianPose>("attractor").is_empty());
+  EXPECT_TRUE(ds->get_base_frame().is_empty());
+  ds->set_parameter(make_shared_parameter("attractor", attractor));
+  EXPECT_FALSE(ds->get_parameter_value<CartesianPose>("attractor").is_empty());
+  EXPECT_FALSE(ds->get_base_frame().is_empty());
   // when attractor was set without a base frame, expect base frame to be identity with name / reference_frame of attractor
-  EXPECT_EQ(ds.get_base_frame().get_name(), attractor.get_reference_frame());
-  EXPECT_EQ(ds.get_base_frame().get_reference_frame(), attractor.get_reference_frame());
-  EXPECT_EQ(ds.get_base_frame().get_transformation_matrix(), Eigen::Matrix4d::Identity());
+  EXPECT_EQ(ds->get_base_frame().get_name(), attractor.get_reference_frame());
+  EXPECT_EQ(ds->get_base_frame().get_reference_frame(), attractor.get_reference_frame());
+  EXPECT_EQ(ds->get_base_frame().get_transformation_matrix(), Eigen::Matrix4d::Identity());
 
-  ds = Linear<CartesianState>();
+  ds = DynamicalSystemFactory<CartesianState>::create_dynamical_system(
```

Comment:
if you're reconstructing the DS again here, then it would be cleaner to have the next block as a separate TestCase